client/src/components/Donation/DonateForm.tsx

```diff
@@ -242,88 +247,64 @@ class DonateForm extends Component<DonateFormProps, DonateFormState> {
     );
   }
 
-  renderDurationAmountOptions() {
-    const { donationAmount, donationDuration, processing } = this.state;
-    const { t } = this.props;
+  resetDonation() {
+    return this.props.updateDonationFormState({ ...defaultDonationFormState });
+  }
 
-    return !processing ? (
-      <div>
-        <h3>{t('donate.gift-frequency')}</h3>
-        <Tabs
-          activeKey={donationDuration}
-          animation={false}
-          bsStyle='pills'
-          className='donate-tabs'
-          id='Duration'
-          onSelect={this.handleSelectDuration}
-        >
-          {(Object.keys(this.durations) as ['month' | 'onetime']).map(
-            duration => (
-              <Tab
-                eventKey={duration}
-                key={duration}
-                title={this.durations[duration]}
-              >
-                <Spacer />
-                <h3>{t('donate.gift-amount')}</h3>
-                <div>
-                  <ToggleButtonGroup
-                    animation={`false`}
-                    className='amount-values'
-                    name='amounts'
-                    onChange={this.handleSelectAmount}
-                    type='radio'
-                    value={this.getActiveDonationAmount(
-                      duration,
-                      donationAmount
-                    )}
-                  >
-                    {this.renderAmountButtons(duration)}
-                  </ToggleButtonGroup>
-                  <Spacer />
-                  {this.renderDonationDescription()}
-                </div>
-              </Tab>
-            )
-          )}
-        </Tabs>
+  paymentButtonsLoader() {
+    return (
+      <div className=' donation-completion donation-completion-loading'>
+        <Spinner
+          className='script-loading-spinner'
+          fadeIn='none'
+          name='line-scale'
+        />
       </div>
-    ) : null;
+    );
   }
 
-  hideAmountOptionsCB(hide: boolean) {
-    this.setState({ processing: hide });
+  renderCompletion(props: {
+    processing: boolean;
+    redirecting: boolean;
+    success: boolean;
+    error: string | null;
+    reset: () => unknown;
+  }) {
+    return <DonateCompletion {...props} />;
   }
 
-  renderDonationOptions() {
+  renderButtonGroup() {
+    const { donationAmount, donationDuration } = this.state;
     const {
+      donationFormState: {
+        loading: { stripe, paypal }
+      },
       handleProcessing,
-      isSignedIn,
       addDonation,
-      t,
       defaultTheme,
-      theme
+      theme,
+      t,
+      isMinimalForm
     } = this.props;
-    const { donationAmount, donationDuration } = this.state;
+    const paymentButtonsLoading = stripe && paypal;
```

Comment:
```suggestion
    const paymentButtonsLoading = loading.stripe && loading.paypal;
```
With this change, the nature of `stripe` and `paypal` are obvious.